core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/metrics/BytesReadTrackingInputStream.java

```diff
@@ -68,6 +72,10 @@ public int read(byte[] b) throws IOException {
     private void updateBytesRead(long read) {
         if (read > 0) {
             bytesRead.addAndGet(read);
+
+            if (progressUpdaterInvoker != null && progressUpdaterInvoker.progressUpdater() != null) {
```

Comment:
This seems a bit code smell since this class requires to know the existence of progress updater before invoking `updateBytesTransferred`.

Can we just call `progressUpdaterInvoker.updateBytesTransferred(bytesRead.get());` and update `progressUpdaterInvoker` to do no-op if progressUpdater is null?

Reply:
done. good idea, included a No-op progressUpdater